src/users.js

```diff
@@ -1,20 +1,35 @@
 import { copyItem, getRandomIndex } from "./helper-functions";
 
+const getRandomUser = users => {
+  return users[getRandomIndex(users)]
+}
+
 const addUniqueRecipes = (userRecipes, newRecipe) => {
   if(!userRecipes.find(item => item.id === newRecipe.id)) userRecipes.push(newRecipe)
 };
 
-const addRecipeToCook = (user, recipe) => {
+const addRecipe = (userRecipes, newRecipe) => {
+  if (!userRecipes) userRecipes = [newRecipe];
+  addUniqueRecipes(userRecipes, newRecipe);
+  return userRecipes;
+}
+
+const removeRecipe = (userRecipes, recipeToRemove) => {
+  let found = userRecipes.find(recipe => recipe.id === recipeToRemove.id);
+  if (found) userRecipes.splice(userRecipes.indexOf(found), 1);
+  return userRecipes;
+}
+
+const updateRecipesToCook = (user, recipe, change) => {
   const updatedUser = copyItem(user);
   let recipes = updatedUser.recipesToCook;
-  if (!recipes) recipes = [recipe];
-  addUniqueRecipes(recipes, recipe)
-  updatedUser.recipesToCook = recipes;
+  let recipeUpdate = {
+    add: () => addRecipe(recipes, recipe),
+    remove: () => removeRecipe(recipes, recipe)
+  }
+  updatedUser.recipesToCook = recipeUpdate[change]();
```

Comment:
omg what is happening here

Reply:
Instead of repeating the logic of making this a pure function in two places I put both the functions for adding and removing recipes in one function, inside an object so then the parent function can have a command passed through and execute accordingly

Reply:
holy crap i see now - this is cool